### ISSUE TITLE
Incorrect Nonce error in class

### ISSUE THREAD

**Issue by kkmuffme** [WordPress-Coding-Standards/WordPress-Coding-Standards#1694]
## Bug Description
When a class is wrapped/loaded by a function, it incorrectly reports that nonce verification is missing (even though it is there. Works fine in normal class/extends & function though)

## Minimal Code Snippet

```php
<?php
/**
 * test
 *
 * @return void
 */
function test() {
	if ( isset( $_POST['my_nonce'] ) && wp_verify_nonce( sanitize_text_field( wp_unslash( $_POST['my_nonce'] ) ), 'the_nonce' ) ) {
		if ( isset( $_POST['hello'] ) ) {
			echo 'world';
		}
	}
}

/**
 * Load function
 *
 * @return mixed
 */
function test_class_loader() {
	if ( !class_exists( 'Test_Method' ) ) {
		/**
		 * Extend shipping method class
		 */
		class Test_Method extends WC_Shipping_Method {
			/**
			 * construct it
			 *
			 * @return mixed
			 */
			public function __construct() {
				// do something
			}

			/**
			 * test
			 *
			 * @return void
			 */
			public function test() {
				if ( isset( $_POST['my_nonce'] ) && wp_verify_nonce( sanitize_text_field( wp_unslash( $_POST['my_nonce'] ) ), 'the_nonce' ) ) {
					if ( isset( $_POST['hello'] ) ) {
						echo 'world';
					}
				}
			}
		}
	}
}

/**
 * Extend shipping method class
 */
class Alternate_Test_Method extends WC_Shipping_Method {
	/**
	 * construct it
	 *
	 * @return mixed
	 */
	public function __construct() {
		// do something
	}

	/**
	 * test
	 *
	 * @return void
	 */
	public function alternate_test() {
		if ( isset( $_POST['my_nonce'] ) && wp_verify_nonce( sanitize_text_field( wp_unslash( $_POST['my_nonce'] ) ), 'the_nonce' ) ) {
			if ( isset( $_POST['hello'] ) ) {
				echo 'world';
			}
		}
	}
}
```

## Error Code
```
 41 | ERROR   | Processing form data without nonce verification. (WordPress.Security.NonceVerification.Missing)
 41 | ERROR   | Processing form data without nonce verification. (WordPress.Security.NonceVerification.Missing)
 42 | ERROR   | Processing form data without nonce verification. (WordPress.Security.NonceVerification.Missing)
```

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.0/7.3
| PHP_CodeSniffer version | 3.4.2
| WPCS version            | 2.1.0 (also tested with 2.0.0, but same problem there)
| WPCS install type       | Composer global

## Tested Against `develop` branch?
- [ ] I have verified the issue still exists in the `develop` branch of WPCS

**Comment by dingo-d** [comment]
Why would you define a class inside a function 😶? 

Also, I'd use early return strategy here instead of nested `if`s.

**Comment by kkmuffme** [comment]
The code is just complete pseudo to reproduce the error easily & showcase the 3 scenarios, of which 2 work as expected.

The reason the class is wrapped in a function is that I want to extend the class conditionally, thus using the wrapper. e.g.
```
if ( $something === true ) {
    add_action( 'some_action', 'test_class_loader' );
}
```

**Comment by jrfnl** [comment]
@kkmuffme Thanks for reporting this. Verified and this is also the case when a superglobal is used in a closure nested within a function. PR upcoming.
